examples/tv-casting-app/tv-casting-common/core/CastingPlayerDiscovery.h

```diff
@@ -0,0 +1,118 @@
+/*
+ *
+ *    Copyright (c) 2023 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include "Types.h"
+#include "CastingPlayer.h"
+
+#include <vector>
+#include <controller/CHIPCommissionableNodeController.h>
+#include <controller/DeviceDiscoveryDelegate.h>
+
+namespace matter {
+namespace casting {
+namespace core {
+
+/**
+ * @brief Represents CastingPlayerDiscovery state.
+ *
+ */
+enum CastingPlayerDiscoveryState
+{
+    DISCOVERY_NOT_READY,  // Default state, mClientDelegate isn't initialized
+    DISCOVERY_READY,      // After SetDelegate and before StartDiscovery, mClientDelegate is initialized
+    DISCOVERY_RUNNING,    // After StartDiscovery success
+};
+
+class DLL_EXPORT DiscoveryDelegate
```

Comment:
Describe what this DiscoveryDelegate does. You can follow [https://github.com/sharadb-amazon/connectedhomeip/blob/casting/examples/tv-casting-app/APIs.md](https://github.com/sharadb-amazon/connectedhomeip/blob/casting/examples/tv-casting-app/APIs.md) to get the description to add here.

Reply:
added